html/assets/js/tools/theme.js

```diff
@@ -2,21 +2,28 @@
 const setTheme = (theme) => {
     document.documentElement.className = theme;
     localStorage.setItem('theme', theme);
+
+    if (typeof setDisplaySettings === "function")
+      setDisplaySettings(theme);
```

Comment:
Don't use setDisplaySettings here. By passing a undefined value of kanji_speed (second parameter), it will reset this setting to [1]. 

Directly use _Util.setMdlCheckboxState("use_dark_mode_settings", theme === "dark");_ instead.
By moving  _Util.setMdlCheckboxState_ from utils2.js into utils.js, you remove the typeof check. 
(utils.js is loaded before theme.js)

Also, if you change utils.js you can do me a favor here by removing _Util.hexToRgb_ which is currently unused.

Reply:
You're right about `setDisplaySettings`. However, using `Util.setMdlCheckboxState` in `theme.js` causes the element returned by the JQuery function to be undefined, independent of whether the function is implemented in `utils.js` or `utils2.js`. This went against my idea of changing as little as possible.

I could get `setDisplaySettings` to fetch `kanji_speed` from `localStorage` if it's not passed as a parameter to the function, or just not touch the value. 

If however the use of `setDisplaySettings` is discouraged there by design instead of due to its effect on `kanji_speed`, I'll continue looking for a solution.

Reply:
It probably fails due to MDL not being set up when setTheme is called the first time.
You could either check if MDL has been set up or check if the query in `Util.setMdlCheckboxState ` returned an item before checking / unchecking it.